src/livestatus.c

```diff
@@ -0,0 +1,479 @@
+      .requests_rate = strtod(fields[16], NULL),
+      .service_checks = strtoull(fields[17], NULL, 10),
+      .service_checks_rate = strtod(fields[18], NULL),
+  };
+
+  return status;
+} /* livestatus_status_t c_to_livestatus_status */
+
+// Borrowed from https://stackoverflow.com/a/29378380
+static long ls_strtol_subrange(const char *s, char **endptr, int base, long min,
+                               long max) {
+  long y = strtol(s, endptr, base);
+  if (y > max) {
+    errno = ERANGE;
+    return max;
+  }
+  if (y < min) {
+    errno = ERANGE;
+    return min;
+  }
+  return y;
+} /* static long ls_strtol_subrange */
+
+// Borrowed from https://stackoverflow.com/a/29378380
+static int ls_strtoi(const char *s, char **endptr, int base) {
+#if INT_MAX == LONG_MAX && INT_MIN == LONG_MIN
+  return (int)strtol(s, endptr, base);
+#else
+  return (int)ls_strtol_subrange(s, endptr, base, INT_MIN, INT_MAX);
+#endif
+} /* static int ls_strtoi */
+
+static int ls_init(void) {
+  livestatus_obj.socket_file = "/var/cache/naemon/live";
+  livestatus_obj.max_retry = 20;
+  livestatus_obj.backoff_sec = 1;
+
+  return 0;
+} /* int ls_init */
+
+static int ls_config(const char *key, const char *value) /* {{{ */
+{
+  if (strcasecmp("LivestatusSocketFile", key) == 0) {
+    char *socket_file = strdup(value);
+    if (socket_file == NULL) {
+      ERROR("livestatus plugin: strdup failed: %s", STRERRNO);
+      return -1;
+    }
+
+    livestatus_obj.socket_file = socket_file;
+  }
+
+  if (strcasecmp("OnFailureMaxRetry", key) == 0) {
+    livestatus_obj.max_retry = ls_strtoi(value, NULL, 10);
+    if (errno != 0) {
+      ERROR("livestatus plugin: strtoi failed for OnFailureMaxRetry: %s",
+            STRERRNO);
+      return -1;
+    }
+  }
+
+  if (strcasecmp("OnFailureBackOffSeconds", key) == 0) {
+    livestatus_obj.backoff_sec = ls_strtoi(value, NULL, 10);
+    if (errno != 0) {
+      ERROR("livestatus plugin: strtoi failed for OnFailureBackOffSeconds: %s",
+            STRERRNO);
+      return -1;
+    }
+
+    return 0;
+  } /* static int ls_config */
+
+  static int unix_connect(const char *sockfile, int *sockfd) {
+    struct sockaddr_un sun;
+    int rc = -1;
+    int sfd = -1;
+
+    sfd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (sfd < 0) {
+      ERROR("livestatus plugin: creating client socket: %s", STRERRNO);
+      return -1;
+    }
+
+    memset(&sun, 0x0, sizeof(sun));
+    sun.sun_family = AF_UNIX;
+    strcpy(sun.sun_path, sockfile);
+
+    do {
+      rc = connect(sfd, (struct sockaddr *)&sun, sizeof(sun));
+    } while (errno == EINTR);
+
+    if (rc < 0) {
+      ERROR("livestatus plugin: connect to socket file %s: %s", sockfile,
+            STRERRNO);
+      return -1;
+    }
+
+    *sockfd = sfd;
+
+    return 0;
+  } /* unix_connect */
+
+  static int ls_parse(const char *lresponse, livestatus_status_t *lstatus) {
+    int rc = -1;
+    int i = 0;
+    char **fields = NULL;
+    char *pchar = NULL;
```

Comment:
Please define variables on first use, here and below.

https://github.com/collectd/collectd/blob/master/docs/review_comments.md#define-variables-on-first-use

Reply:
Should be fixed now.